### PR TITLE
Update to Kotlin 1.9.0:

### PR DESCRIPTION
**Pull request by sandwwraith**
- Apply @ExperimentalNativeApi on WeakReference usages
- Drop Legacy JS compiler as it is deprecated
- Update API Dump with .entries enum properties
- Update yarn.lock

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -27,7 +27,7 @@ kover.enabled=true
 kotlin.mpp.enableCompatibilityMetadataVariant=true
 kotlin.mpp.stability.nowarn=true
 
-kotlin.js.compiler=both
+kotlin.js.compiler=ir
```
Comment (JakeWharton) [inline gradle.properties:30]: This can be straight-up removed, right? IR is the default and so you don't need to specify this anymore.

Comment (sandwwraith) [reply gradle.properties:30]: I guess so, but there's no harm in explicitly mentioning that

**Review by shanshin — COMMENTED** [review]
It seems to me that we should also remove functions related to legacy JS (like `noJsLegacy`, `isJsLegacy` etc), since they are no longer relevant and can only confuse

File: gradle.properties (hunk 1 above)
Comment (qwwdfsad) [reply gradle.properties:30]: I'd suggest removing it, it's really tough to keep track of all the properties later on

**Review by qwwdfsad — CHANGES_REQUESTED** [review]
We have quite a few `legacy`-related quirks -- `isLegacyBackend`, `compileTestKotlinJsLegacy` etc. Would be nice to get rid of them.

Also, please file an issue to support SubclassOptInRequired

**Review by shanshin — APPROVED** [review]

**Comment by sandwwraith** [comment]
@qwwdfsad

**Pull request merged**